al_bench/model.py

```diff
@@ -136,31 +142,107 @@ def get_dataset_handler(self):
 
     import keras
 
+    # Does it matter that this GenericModelHandler code requires `import tensorflow`
+    # because of the superclass of GenericModelHandler.CustomCallback?!!!
+
+    # Does it matter that this GenericModelHandler code requires `import torch`` because
+    # of the use of torch.utils.tensorboard.SummaryWriter in
+    # GenericModelHandler.CustomCallback.write_some_log_for_tensorboard?!!!
+
     class CustomCallback(keras.callbacks.Callback):
         def __init__(self):
             super(GenericModelHandler.CustomCallback, self).__init__()
             self.reset_log()
+            self.training_size = 0
 
         def reset_log(self):
             self.log = list()
 
         def get_log(self):
             return self.log
 
+        def write_some_log_for_tensorboard(
+            self, model_step, y_dictionary, x_key, *args, **kwargs
+        ):
+            from torch.utils.tensorboard import SummaryWriter
+
+            if self.log is None:
+                return False
+            with SummaryWriter(*args, **kwargs) as writer:
+                beginning = datetime.utcfromtimestamp(0)
+                for entry in self.log:
+                    if entry["model_step"] != model_step:
+                        continue
+                    logs = entry["logs"]
+                    if logs is None:
+                        continue
+                    utc_seconds = (entry["utcnow"] - beginning).total_seconds()
+                    x_value = entry[x_key]
+                    for key in y_dictionary.keys():
```

Comment:
Doesn't the standard dictionary iterator iterate over keys?  That is, I think `.keys()` is unnecessary here and in the line below.

Reply:
Thank you.  I will change those.